### PR TITLE
Pull request for gslab-econ/gslab_python#33: Template changes for release.py changes

### PR DESCRIPTION
**Pull request by M-R-Sullivan**
I have edited large_template to work with the release protocol developed in gslab-econ/gslab_python#33, which is under review at gslab-econ/gslab_python#36.

### CODE REVIEW

**Comment by arosenbe** [comment]
On hold while we finish up congress text. Sorry @M-R-Sullivan.

**Comment by arosenbe** [comment]
On hold while we finish up congress text. Sorry @M-R-Sullivan.

**Comment by arosenbe** [comment]
@M-R-Sullivan, the code looks fine, but I can't run the repo. Can you look into this?

**Comment by M-R-Sullivan** [comment]
@arosenbe - I can! I'll let you know what I find.

**Comment by arosenbe** [comment]
The issue was with my R after updating to Sierra, I had to retap homebrew/science and then reinstall r. Everything's hunky-dory now.

**Comment by arosenbe** [comment]
@M-R-Sullivan, as discussed, I'm getting this error message.
```
(gslab_python-issue33-release) template$ python -m gslab_scons.release version=0.0.0
WARNING: `scons` has run since your latest git commit.

Would you like to continue anyway? (y|n)
y
Enter a GitHub token and then press enter: 
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/gslab_scons/release.py", line 116, in <module>
    zip_release       = zip_release)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/gslab_scons/_release_tools.py", line 52, in release
    tag_name_index = json_split.index('"tag_name":"%s"' % tag_name)
ValueError: '"tag_name":"0.0.0"' is not in list
```

**Comment by M-R-Sullivan** [comment]
@arosenbe - I just tried running `python -m gslab_scons.release version=0.0.0` with the latest version of `gslab_python` from branch `issue33-release` and it worked. Could you try again with this version and let me know if you get this error again? This isn't time sensitive.

**Comment by arosenbe** [comment]
Thanks @M-R-Sullivan, I tried running the code and got the error below. I recognize that this is because my current run isn't after a cache, but we should fix the Exception message, and maybe understand why it didn't print as intended.
```
Error: up_to_date(mode = scons) must be run on a SCons directory.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/Users/arosenbe/Desktop/gslab_python/gslab_scons/release.py", line 14, in <module>
    if not up_to_date(mode = 'scons'):
  File "gslab_scons/_release_tools.py", line 205, in up_to_date
    raise ReleaseError('up_to_date(mode = scons) must be run on a '
gslab_scons._exception_classes.ReleaseError
```

**Comment by M-R-Sullivan** [comment]
@arosenbe - agreed. I'll check that out.

**Comment by arosenbe** [comment]
Also, I don't understand why I got that error. Maybe you could explain?

**Comment by M-R-Sullivan** [comment]
@arosenbe - I could! If you have time today, I'd be happy to take a look at what's going wrong for you, as I didn't get that error when I ran gslab_scons.release in template.

**Comment by arosenbe** [comment]
It works!

File: SConstruct [hunk 1]
```diff
@@ -36,21 +27,9 @@ Export('env')
 SConscript('source/data/SConscript') 
 
 # Additional mode options
-if mode in ['cache', 'release']:
+if mode == 'cache':
```
Comment (arosenbe) [inline SConstruct]: Could we change this syntax to 
```python
if mode is 'cache'
```
I think that's the way real people do it.

**Comment by M-R-Sullivan** [comment]
@arosenbe - I responded to your `==` vs `is` point. As per our discussion, I'm merging now.

File: SConstruct (hunk 1 above)
Comment (M-R-Sullivan) [inline SConstruct]: @arosenbe - I've been using `is` in the place of `==` for string comparison, and I've found that `is` often returns `False` when `==` returns `True`. I just tried it out in the template and when `mode=cache` is specified, `mode is 'cache'` returns `False` wereas `mode =='cache'` is `True`. Hence I am reversing this change in `master`.

**Comment by arosenbe** [comment]
Sounds fine. Sorry to bring you down that wrong path.

**Pull request merged**